swap_meet/vendor.py

```diff
@@ -1,2 +1,77 @@
 class Vendor:
-    pass
+    def __init__(self,inventory=None):
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+        # if inventory != None else inventory = []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False    
+
+    def get_by_category(self, category):
+        result = []
+        for each in self.inventory:
+            if each.category == category:
+                result.append(each)
+        return result
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in vendor.inventory:
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            vendor.inventory.remove(their_item)
+            vendor.inventory.append(my_item)
+            return True
+        else:
+            return False
```

Comment:
Every `if` conditional doesn't need to be accompanied by an `else`. When we get into web development, else clauses are useful in scenarios where we need a 'catch all' such as handling gibberish a user enters into a form. 

Let's say a user was trying to enter their email in a form.. if the user enters `!@#AFS12334` into the form, we can have conditionals describe the input as needing to be a string type, containing alphanumeric values, etc. and in case the input doesn't cover all the required checks we can use  `else: return "Please enter a valid email"` as a catch all for invalid inputs.

Reply:
We can make the appending and removing logic here more 'focused' in this function by using a guard clause:

```suggestion
    def swap_items(self, vendor, my_item, their_item):
        if my_item not in self.inventory and their_item not in vendor.inventory:
            return False
        self.inventory.remove(my_item)
        self.inventory.append(their_item)
        vendor.inventory.remove(their_item)
        vendor.inventory.append(my_item)
        return True
```